lib/agent/utils/storage.js

```diff
@@ -1,116 +1,171 @@
-var fs = require('fs');
-var db, db_path;
+var fs = require('fs'),
+    sqlite3 = require('sqlite3').verbose();
 
-var load = function(cb) {
-  if (db)
-    return cb();
+var db_path, db_type, db_comm;
+
+var db_commands = {
+  CREATE: function(type) {return "CREATE TABLE IF NOT EXISTS " + type + " (" + singular(type) + " text)"},
+  INSERT: function(type) {return "INSERT INTO " + type + " VALUES (?)"},
+  SELECT: function(type) {return "SELECT * FROM " + type},
+  DELETE: function(type) {return "DELETE FROM " + type}
+}
+
+var singular = function(db_type) {
+  return db_type.substring(0, db_type.length - 1);
+}
 
-  if (!db_path)
-    return cb(new Error('Invalid path'));
+var check_type = function(key, cb) {
+  if (key.indexOf('start') != -1) return cb(null, 'commands');
+  else if (key.length == 32 && key == key.toUpperCase()) return cb(null, 'files');
+  else return cb(new Error("Not commands or files"))
+}
 
-  fs.readFile(db_path, 'utf8', function(err, data) {
+var load = function(type, cb) {
+  var db;
+  if (!db_path) {
+    return cb(new Error('Invalid path'), null);
+  }
+
+  db_comm.all(db_commands.SELECT(type), function(err, rows) {
     if (err) {
       if (err.code != 'ENOENT')
-        return cb(err); // only return error if different from ENOENT
+        return cb(err, null);
 
       db = {};
-      return cb();
+      return cb(err, db);
     }
 
     try {
-      db = JSON.parse(new Buffer(data, 'base64').toString());
+      var db1 = {};
```

Comment:
this hole logic could be simplified if the data is already stored in the format you need it